src/mongo/client/native_sasl_client_session.cpp

```diff
@@ -38,7 +38,7 @@
 namespace mongo {
 namespace {
 
-SaslClientSession* createNativeSaslClientSession(const std::string mech) {
+SaslClientSession* createNativeSaslClientSession(const std::string& mech) {
```

Comment:
Polymorphism. SaslClientSession::create is set to point to this function. Another implementation, createCyrusSaslClientSession, uses the parameter.